From 5b78e31d0aeb0cfcf2d3e108326c660d3cfd7aa1 Mon Sep 17 00:00:00 2001
From: Aikar <aikar@aikar.co>
Date: Mon, 6 Nov 2017 21:10:01 -0500
Subject: [PATCH] API to get a BlockState without a snapshot

This allows you to get a BlockState without creating a snapshot, operating
on the real tile entity.

This is useful for where performance is needed

diff --git a/src/main/java/org/bukkit/block/Block.java b/src/main/java/org/bukkit/block/Block.java
index 1aa3373f..9aab043c 100644
--- a/src/main/java/org/bukkit/block/Block.java
+++ b/src/main/java/org/bukkit/block/Block.java
@@ -242,6 +242,15 @@ public interface Block extends Metadatable {
      */
     BlockState getState();
 
+    // Paper start
+    /**
+     * @see #getState() optionally disables use of snapshot, to operate on real block data
+     * @param useSnapshot if this block is a TE, should we create a fully copy of the TileEntity
+     * @return BlockState with the current state of this block
+     */
+    BlockState getState(boolean useSnapshot);
+    // Paper end
+
     /**
      * Returns the biome that this block resides in
      *
-- 
2.18.0

